packages/jest-core/src/cli/index.ts

```diff
@@ -176,17 +196,34 @@ const runWatch = async (
   globalConfig: Config.GlobalConfig,
   outputStream: NodeJS.WriteStream,
   hasteMapInstances: Array<HasteMap>,
+  filter?: Filter,
 ) => {
   if (hasDeprecationWarnings) {
     try {
       await handleDeprecationWarnings(outputStream, process.stdin);
-      return watch(globalConfig, contexts, outputStream, hasteMapInstances);
+      return watch(
+        globalConfig,
+        contexts,
+        outputStream,
+        hasteMapInstances,
+        undefined,
+        undefined,
+        filter,
```

Comment:
@SimenB we need to convert this to an object finally :D

Reply:
Agree... I'm happy to handle that in a second PR if we want to do it now.

Reply:
Awesome! looks like this won't break public API for `jest-core` so happy to land that before next major